### PR TITLE
theft: init at 0.4.3

### PR DESCRIPTION
**Pull request by kquick**
###### Motivation for this change

Package is not currently in nixpkgs.

Depends on pull request #30565

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
     *N/A*
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
     *N/A*
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
     *N/A*
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/libraries/theft/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  propagatedBuildInputs = [ ];
+
+  buildInputs = [ ];
+
+  installFlags = [ "PREFIX=$(out)" ];
+  postInstall = ''
+      install -m644 vendor/greatest.h $out/include/
+    '';
+  
+  meta = {
```
Comment (peterhoeg) [inline pkgs/development/libraries/theft/default.nix]: If you change this to ```meta = with stdenv.lib; {``` you can get rid of all the ```stdenv.lib``` stuff below.

File: pkgs/development/libraries/theft/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+  src = fetchFromGitHub {
+    owner = "silentbicycle";
+    repo = "theft";
+    rev = "v${version}";
+    sha256 = "1ibh8np12lafnrsrvjbbzlyq45zq654939x0y22vdnc6s8dpbhw4";
+  };
+
+  preConfigure = "patchShebangs ./scripts/mk_bits_lut";
+  # doCheck = true;
```
Comment (peterhoeg) [inline pkgs/development/libraries/theft/default.nix]: Any particular reason checks are not enabled? Please document that.

File: pkgs/development/libraries/theft/default.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  
+  propagatedBuildInputs = [ ];
```
Comment (peterhoeg) [inline pkgs/development/libraries/theft/default.nix]: Get rid of this

File: pkgs/development/libraries/theft/default.nix [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+  buildInputs = [ ];
```
Comment (peterhoeg) [inline pkgs/development/libraries/theft/default.nix]: Get rid of this

File: pkgs/development/libraries/theft/default.nix [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+  installFlags = [ "PREFIX=$(out)" ];
```
Comment (peterhoeg) [inline pkgs/development/libraries/theft/default.nix]: This should be passed to the configure phase automatically. Are you sure this is needed?

Comment (kquick) [reply pkgs/development/libraries/theft/default.nix]: This package does not use autoconf or any other configuration option.  The --prefix option is supplied during the configurePhase, but no corresponding specification is normally provided during the installPhase.  The Makefile for this project assumes the older-style convention of specifying PREFIX when running the install target, which is what the above is intended to accomplish.    This is necessary and works correctly in my testing of this derivation.

File: pkgs/development/libraries/theft/default.nix (hunk 4 above)
Comment (kquick) [reply pkgs/development/libraries/theft/default.nix]: removed

File: pkgs/development/libraries/theft/default.nix (hunk 3 above)
Comment (kquick) [reply pkgs/development/libraries/theft/default.nix]: removed

File: pkgs/development/libraries/theft/default.nix (hunk 2 above)
Comment (kquick) [reply pkgs/development/libraries/theft/default.nix]: checks are now enabled

File: pkgs/development/libraries/theft/default.nix (hunk 1 above)
Comment (kquick) [reply pkgs/development/libraries/theft/default.nix]: Understood, but are you indicating this as a helpful hint, or is this a requirement for acceptance of this package?  If it's not required, my preference would be for the current explicit form, but I will make this change if required.

**Comment by kquick** [comment]
@peterhoeg, thank you for the initial review.  I have update the pull request and would appreciate your feedback and re-review.

File: pkgs/development/libraries/theft/default.nix [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+  preConfigure = "patchShebangs ./scripts/mk_bits_lut";
+
+  doCheck = true;
+  checkTarget = "test";
+  
+  installFlags = [ "PREFIX=$(out)" ];
+  postInstall = ''
+      install -m644 vendor/greatest.h $out/include/
+    '';
```
Comment (peterhoeg) [inline pkgs/development/libraries/theft/default.nix]: indentation is off here

File: pkgs/development/libraries/theft/default.nix (hunk 1 above)
Comment (peterhoeg) [reply pkgs/development/libraries/theft/default.nix]: It's pretty subjective, so no, not a requirement. I however, find it easier to read.

**Comment by peterhoeg** [comment]
Thanks @kquick !

**Pull request merged**